src/OSPSuite.Infrastructure.Import/Core/DataSource.cs

```diff
@@ -263,16 +263,10 @@ private ParseErrors validateUnitsSupportedAndSameDimension(Cache<string, ColumnI
                   //if unit comes from a column
                   if (column.Key.Column.Dimension == null)
                   {
-                     var firstNonEmptyUnit = column.Value.FirstOrDefault(x => !string.IsNullOrEmpty(x.Unit));
+                     var firstValueWithNonEmptyUnit = column.Value.FirstOrDefault(x => !string.IsNullOrEmpty(x.Unit));
+                     var firstNonEmptyUnit = firstValueWithNonEmptyUnit == null ? "" : firstValueWithNonEmptyUnit.Unit;
```

Comment:
FYI you can write it as 

`var firstNonEmptyUnit = firstValueWithNonEmptyUnit?.Unit??"";`

Reply:
I like the suggestion. Implemented.